pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go

```diff
@@ -421,6 +428,7 @@ func (sf *streamIngestionFrontier) maybeUpdatePartitionProgress() error {
 	partitionProgress := sf.partitionProgress
 
 	sf.lastPartitionUpdate = timeutil.Now()
+	sf.persistedHighWater = f.Frontier()
```

Comment:
Should we move this until after we've updated the job so that it is only saved after we've persisted the frontier?

Also, while we persist the frontier checkpoint here, we haven't actually persisted the high watermark yet right?  We don't persist the high watermark until we send the row back to the row receiver in the distsql flow. Perhaps it doesn't matter in practice since individual partitions will re-initialize their frontier from the checkpoint.  But, I kinda wonder if we should move the high watermark persistence into the frontier processor so that we can _know_ when it is actually persisted.  Wonder if @miretskiy has thoughts

Reply:
We should move it to frontier processor; I suspect these are remnants of old design where aggregators were a lot more stateful.